tools/load_system_tablets.cc

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "tools/load_system_tablets.hh"
+
+#include <algorithm>
+#include <seastar/core/thread.hh>
+#include <seastar/util/closeable.hh>
+
+#include "log.hh"
+#include "partition_slice_builder.hh"
+#include "data_dictionary/keyspace_metadata.hh"
+#include "db/cql_type_parser.hh"
+#include "db/system_keyspace.hh"
+#include "mutation/mutation.hh"
+#include "readers/combined.hh"
+#include "reader_concurrency_semaphore.hh"
+#include "tools/read_mutation.hh"
+#include "tools/sstable_manager_service.hh"
+#include "types/list.hh"
+#include "types/tuple.hh"
+
+namespace {
+
+logging::logger logger{"load_sys_tablets"};
+
+
+future<std::filesystem::path> get_table_directory(std::filesystem::path scylla_data_path,
+                                                  std::string_view keyspace_name,
+                                                  std::string_view table_name) {
```

Comment:
I think this could be shared with `tools/schema_loader.cc` too. Maybe add it to `read_mutation_offline.hh` too, as a related utility.

Reply:
ack.